widget_driver_generator/lib/src/utils/default_return_value_helper.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+/// Class that provides logic to determine default return values.
+class DefaultReturnValueHelper {
+  /// Return the default value written as code inside a String for the passed in [type].
+  /// It is important to call [hasDefaultValueForType] first to make sure a default value exists.
+  /// Otherwise the code generation might generate faulty code.
+  static String getDefaultValueFor(DartType type) {
+    if (type.element is EnumElement) {
+      // We can safely take the first enum value since dart enforces for enums to have at least one value.
+      return "${_getSanitizedTypeName(type.toString())}.values[0]";
+    }
+    if (type is RecordType) {
+      return _getDefaultValueForRecord(type);
+    }
+    if (type is InterfaceType && type.element is ClassElement && type.element.name == "Future") {
+      final defaultValue = getDefaultValueFor(type.typeArguments.first);
+      return "Future.value($defaultValue)";
+    } else {
+      return _getDefaultValueForTypeName(type.toString());
+    }
+  }
+
+  /// Returns `true` if a default value exists for the passed in [type].
+  /// If it returns `false`, [getDefaultValueFor] might not return a proper value and should not be used.
+  static bool hasDefaultValueForType(DartType type) {
+    if (type.element is EnumElement) {
+      return true;
+    }
+    // RecordTypes are tuples (value1, value2) that were introduced with dart 3.0 and are a edge case here
+    if (type is RecordType) {
+      return _hasDefaultValueForRecord(type);
+    }
+    var typeName = type.toString();
+    if (type is InterfaceType && type.element is ClassElement && type.element.name == "Future") {
+      typeName = type.typeArguments.first.toString();
+    }
+    return _hasDefaultValueForTypeName(typeName);
+  }
+
+  // to introduce more types with default values, add them to this map.
+  // All enum values and optional values are already covered and don't have to be put in here.
+  static final Map<String, String> _defaultTypeValueMap = {
```

Comment:
Is it possible to let the consumers somehow add their own definition of defaultTypeValueMaps?
I think you could define like a yaml config file and pass in to the build_runner. 
Or maybe it is possible to let consumers define their own static variable..

So that somehow a consumer could all some predefined value for their own types.. 
Sort of like: `MyCombustionVehicle => MyCombustionVehicle(engine: 0);`

It would make it even more flexible.. But maybe it is not possible

Reply:
That would perhaps be a great future addition. Can we put that into a new feture request instead of this PR? What do you think?

Reply:
yeah sounds good. Let's think about it for future us :D